index/src/executors/bridge_executors/phala_xtransfer.rs

```diff
@@ -0,0 +1,206 @@
+use pink_subrpc::{create_transaction, send_transaction, ExtraParam};
+
+use crate::traits::{common::Error, executor::BridgeExecutor};
+use crate::utils::ToArray;
+use crate::AccountType;
+use alloc::{
+    string::{String, ToString},
+    vec::Vec,
+};
+use scale::Decode;
+
+use xcm::v1::{prelude::*, AssetId, Fungibility, Junctions, MultiAsset, MultiLocation};
+
+pub type Phala2AcalaExecutor = PhalaXTransferExecutor;
+pub type Phala2MoonbeamExecutor = PhalaXTransferExecutor;
+
+#[derive(Clone)]
+pub struct PhalaXTransferExecutor {
+    rpc: String,
+    dest_chain_id: u32,
+    // account32 or account20
+    account_type: AccountType,
+}
+
+#[allow(dead_code)]
+impl PhalaXTransferExecutor {
+    pub fn new(rpc: &str, dest_chain_id: u32, account_type: AccountType) -> Self
+    where
+        Self: Sized,
+    {
+        Self {
+            rpc: rpc.to_string(),
+            dest_chain_id,
+            account_type,
+        }
+    }
+}
+
+impl BridgeExecutor for PhalaXTransferExecutor {
```

Comment:
We'd better call it `PhalaXTransferParaExecutor` bc xtransfer also support transfer to solo chain like Ethereum, which will have different dest location representation